### PR TITLE
XW-3227 | featured video attribution

### PR DESCRIPTION
**Pull request by latata**
https://wikia-inc.atlassian.net/browse/XW-3227

@Wikia/x-wing

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [article-featured-video.js#L1](https://github.com/Wikia/app/blob/49216dbabe51596390e5dce6d6328b7fb90bcdf7/extensions/wikia/ArticleVideo/scripts/article-featured-video.js#L1): Function has 6 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/ArticleVideo/scripts/article-featured-video.js [hunk 1]
```diff
@@ -135,6 +135,15 @@ require(['wikia.window', 'wikia.onScroll', 'wikia.tracker', 'ooyala-player', 'wi
 			}
 		}
 
+		function initAttributionTracking() {
+			$('.attribution-container a').click(function () {
```
Comment (hakubo) [inline extensions/wikia/ArticleVideo/scripts/article-featured-video.js]: $('.attribution-container').find('a') - do we care about optimization like this?

File: extensions/wikia/ArticleVideo/styles/video-attribution.scss [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+@import 'extensions/wikia/DesignSystem/bower_components/design-system/dist/scss/wds-variables/colors';
+
+.attribution-container {
+	@include align-items(center);
+	@include flexbox();
+	@include justify-content(center);
+	border-bottom: 1px solid rgba($color-text, .5);
+	padding: 6px;
+
+	.attribution-avatar {
```
Comment (hakubo) [inline extensions/wikia/ArticleVideo/styles/video-attribution.scss]: does it have to be nested?

File: extensions/wikia/ArticleVideo/styles/video-attribution.scss [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	}
+
+	.attribution-avatar {
+		display: none;
+	}
+
+	.attribution-username, .attribution-icon {
+		color: $wds-color-light-gray;
+	}
+}
```
Comment (hakubo) [inline extensions/wikia/ArticleVideo/styles/video-attribution.scss]: omg! missing line!

File: extensions/wikia/ArticleVideo/scripts/article-featured-video.js (hunk 1 above)
Comment (rogatty) [inline extensions/wikia/ArticleVideo/scripts/article-featured-video.js]: What do you think about scoping it to `.featured-video-wrapper .attribution-container a`?
`.attribution-container` seems pretty generic to me.

File: extensions/wikia/ArticleVideo/styles/video-attribution.scss [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+		color: rgba($color-text, .75);
+	}
+
+	.attribution-icon {
+		line-height: 0;
+		margin-left: 6px;
+	}
+}
+
+.collapsed {
```
Comment (rogatty) [inline extensions/wikia/ArticleVideo/styles/video-attribution.scss]: This is very generic. Why don't we use `.video-attribution__container .is-collapsed` style like in Design System? Let's be consistent.

File: extensions/wikia/ArticleVideo/scripts/article-featured-video.js (hunk 1 above)
Comment (latata) [reply extensions/wikia/ArticleVideo/scripts/article-featured-video.js]: looks like single selectors are faster: https://jsperf.com/jquery-single-selector-vs-finds

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [article-featured-video.js#L1](https://github.com/Wikia/app/blob/61d3250a6a8f9924e179e9c0edd2943d5431f0e0/extensions/wikia/ArticleVideo/scripts/article-featured-video.js#L1): Function has 6 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/ArticleVideo/styles/video-attribution.scss [hunk 5]
```diff
@@ -0,0 +1,44 @@
+@import 'skins/shared/color';
+@import 'skins/shared/mixins/flexbox';
+@import 'extensions/wikia/DesignSystem/bower_components/design-system/dist/scss/wds-variables/colors';
+
+.featured-video-wrapper {
+	.attribution-container {
```
Comment (hakubo) [inline extensions/wikia/ArticleVideo/styles/video-attribution.scss]: how about video-attribution-container
and then
video-attribution-avatar
and so on
then we would not have to nest these that much

File: extensions/wikia/ArticleVideo/styles/video-attribution.scss [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+	.attribution-username, .attribution-icon {
+		color: rgba($color-text, .75);
+	}
+
+	.attribution-icon {
+		line-height: 0;
+		margin-left: 6px;
+	}
+
+	&.collapsed {
```
Comment (hakubo) [inline extensions/wikia/ArticleVideo/styles/video-attribution.scss]: is-collapsed

**Review by rogatty — APPROVED** [review]

**Comment by rybmat** [comment]
closing due to not being able to merge dev to this branch. New PR with the same commits: https://github.com/Wikia/app/pull/12884

**Pull request closed without merging**